### PR TITLE
Remove net40, net45 from NuGet Client SDK

### LINKED ISSUE #11830: Remove net45 and net40 targets from the NuGet build
These are outdated and out of support frameworks.
 
Currently VS 2022 doesn't even give you the option to install the framework components for .NET Framework 4.0 - 4.5.1, which means that you have to jump through hoops to get those targeting packs installed. 

See : https://docs.microsoft.com/en-us/dotnet/framework/install/guide-for-developers

> Starting with Visual Studio 2022, Visual Studio no longer includes .NET Framework components for .NET Framework 4.0 - 4.5.1 because these versions are no longer supported. Visual Studio 2022 and later versions can't build apps that target .NET Framework 4.0 through .NET Framework 4.5.1. To continue building these apps, you can use Visual Studio 2019 or an earlier version.

I understand that there are other reasons for targetting net45 in particular, `NuGet.Packaging.Extraction`, but I think that can be addressed by using older versions of the package that have already been published. 

In particular both @drewgillies and @lukka both ran into issues over the last 2 weeks trying to build the product.

* Note that removing these extra frameworks removes 5 different builds, so it'd speed up the build quite a bit.

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11830

Regression? Last working version:

## Description
Remove target frameworks `net40` and `net45` from our build.  Also remove `NuGet.Packaging.Extraction` since we'll update that code from a servicing branch if needed.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: scripts/utils/EnsureAllPackagesExist.ps1 [hunk 1]
```diff
@@ -27,7 +27,6 @@ Param (
 "NuGet.Localization",
 "NuGet.PackageManagement",
 "NuGet.Packaging.Core",
-"NuGet.Packaging.Extraction",
```
Comment (zivkan) [inline scripts/utils/EnsureAllPackagesExist.ps1:30]: Need to check where this script is used. NuGet.Packaging.Extraction should already be excluded from our nuget.org publishing, but I'm worried this script will break our publishing pipeline for servicing branches.

File: src/NuGet.Core/NuGet.Configuration/Utility/IReadOnlyCollectionUtility.cs [hunk 2]
```diff
@@ -13,20 +13,12 @@ internal static class IReadOnlyCollectionUtility
         internal static IReadOnlyCollection<T> Create<T>(IEqualityComparer<T> comparer, params T[] t)
         {
             var hashSet = new HashSet<T>(t, comparer);
```
Comment (jeffkl) [inline src/NuGet.Core/NuGet.Configuration/Utility/IReadOnlyCollectionUtility.cs]: I almost think we should get rid of this class now that it has no behavior difference.  But it is used in a decent number of places so I'd be fine with leaving it for now, just wanted to throw it out there.

**Review by jeffkl — DISMISSED** [review]
🎉

File: scripts/utils/EnsureAllPackagesExist.ps1 (hunk 1 above)
Comment (nkolev92) [reply scripts/utils/EnsureAllPackagesExist.ps1:30]: Did you find it? :)

**Review by nkolev92 — DISMISSED** [review]

File: scripts/utils/EnsureAllPackagesExist.ps1 (hunk 1 above)
Comment (zivkan) [reply scripts/utils/EnsureAllPackagesExist.ps1:30]: It looks like we'll be ok. The script passes if `NuGet.Packaging.Extraction` (or other nupkgs) exist and are not defined in the list. In theory it means that if a servicing branch fails to create the N.P.E package, then when we publish it won't detect the missing package. But this package isn't pusblished to nuget.org anyway, only to the daily build feed, which happens as part of the build pipeline, not a release pipeline.

**Review by jeffkl — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**